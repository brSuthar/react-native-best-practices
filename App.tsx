/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React, {useEffect, useState} from 'react';
import {
  Alert,
  SafeAreaView,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
} from 'react-native';

function App(): React.JSX.Element {
  const [count, setCount] = useState(0);
  const [text, setText] = useState<string>('');

  return (
    <SafeAreaView style={styles.view}>
      <TouchableOpacity
        testID="button-1"
        onPress={() => {
          setCount(pre => pre + 1);
        }}>
        <Text>Click</Text>
      </TouchableOpacity>
      <Text>Count: {count}</Text>
      <TextInput
        testID="textInput"
        placeholder="username"
        value={text}
        onChangeText={setText}
      />
      <Text>{text}</Text>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  view: {
    height: '100%',
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
  },
});

export default App;
