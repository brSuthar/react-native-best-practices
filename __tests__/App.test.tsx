/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

import {render, fireEvent} from '@testing-library/react-native';
import 'react-test-renderer';

describe('snapshot ', () => {
  it('take snapshot', () => {
    const component = render(<App />);
    expect(component).toMatchSnapshot();
  });
});

describe('test app input', () => {
  it('button test', () => {
    const {getByTestId, getByText} = render(<App />);
    const button = getByTestId('button-1');
    fireEvent.press(button);

    const text = getByText('Count: 1');
    expect(text).toBeTruthy();
  });
  it('input test', () => {
    const {getByTestId} = render(<App />);
    const input = getByTestId('textInput');
    fireEvent.changeText(input, 'Hello');
    expect(input.props.value).toMatch('Hello');
    expect(input.props.placeholder).toBe('username');
    // expect(input.props.keyb)
  });
});
