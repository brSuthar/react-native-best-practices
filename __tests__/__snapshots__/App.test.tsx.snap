// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot  take snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="button-1"
  >
    <Text>
      Click
    </Text>
  </View>
  <Text>
    Count: 
    0
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="username"
    testID="textInput"
    value=""
  />
  <Text />
</RCTSafeAreaView>
`;
